File: src/main/java/axirassa/services/runners/ControllerServiceJob.java

package axirassa.services.runners;

import org.quartz.Job;
import org.quartz.JobExecutionContext;
import org.quartz.JobExecutionException;

import axirassa.ioc.IocStandalone;
import axirassa.services.ControllerService;
import axirassa.services.Service;

public class ControllerServiceJob implements Job {
	@Override
	public void execute (JobExecutionContext context) throws JobExecutionException {
		try {
			Service service = IocStandalone.autobuild(ControllerService.class);
			service.execute();
		} catch (Exception e) {
			throw new JobExecutionException(e);
		}
	}
}

File: src/main/java/axirassa/services/runners/ControllerServiceJob.scala

package axirassa.services.runners

import org.quartz.Job
import org.quartz.JobExecutionContext
import org.quartz.JobExecutionException

import axirassa.ioc.IocStandalone
import axirassa.services.ControllerService
class ControllerServiceJob extends Job {
	override
	def execute (context : JobExecutionContext) {
		try {
			val service = IocStandalone.autobuild(classOf[ControllerService])
			service.execute
		} catch {
			case ex => throw new JobExecutionException(ex)
		}
	}
}
